forte/orbital-helpers/aosubspace.h

```diff
@@ -127,58 +131,51 @@ class AOInfo {
-    // Constructor with list of subspaces
+    /// Constructor using a list of subspaces and atom normals
+    /// @param subspace_str: a list of subspace orbitals, e.g, {"C2", "N", "Fe(3d)", "Mo(3dx2-y2)"}
+    /// @param molecule: a Psi4 Molecule object
+    /// @param basis: a Psi4 basis object, a minimal basis where subspace orbitals are selected
+    /// @param atom_normals: (optional) a map from the atom to its normal
+    /// @param debug_mode: debug mode if True (more printing)
+    ///
+    /// For the argument of atom_normals:
+    ///   An atom is characterized by its atomic number and the relative index.
+    ///   For example, {6, 2} - the third carbon atom of the molecule (index is 0-based)
+    ///
+    ///   This argument is used to make the subspace p orbitals of an atom aligned to the normal:
+    ///   px, py, pz (3 orbitals) -> nx * px + ny * py + nz * pz (1 orbital),
+    ///   where the atom normal is a 3D unit vector (nx, ny, nz).
+    ///
+    ///   If the map is empty, the p orbitals are in the xyz frame defined the molecule.
+    ///   Equivalently, 3 normals are attached to each atom: (1,0,0), (0,1,0), and (0,0,1).
+    ///   As such, we still have a full set of p orbitals in the subspace.
     AOSubspace(std::vector<std::string> subspace_str, std::shared_ptr<psi::Molecule> molecule,
-               std::shared_ptr<psi::BasisSet> basis);
+               std::shared_ptr<psi::BasisSet> basis,
+               std::map<std::pair<int, int>, psi::Vector3> atom_normals = {},
+               bool debug_mode = false);
 
     // ==> User's interface <==
 
-    // Adds a subspace, e.g. add_subspace("C(1s,2s)")
-    void add_subspace(std::string);
-
-    // Compute the AOs in the subspace
-    void find_subspace();
-
-    // Return the index of the AOs that span the subspace selected
-    const std::vector<int>& subspace();
-
-    psi::SharedMatrix build_projector(const std::vector<int>& subspace,
-                                      std::shared_ptr<psi::Molecule> molecule,
-                                      std::shared_ptr<psi::BasisSet> min_basis,
-                                      std::shared_ptr<psi::BasisSet> large_basis);
-
-    /// Return a vector of labels for each atomic orbital.  This function
-    /// accepts
-    /// an optional argument that indicates the formatting that will be fed to
-    /// boost::format.
-    ///
-    /// The field available for printing are:
-    ///   1. Atom number (int)
-    ///   2. Atom label, e.g. "C" (string)
-    ///   3. Atom count, e.g. 3 = third atom of a given kind (int)
-    ///   4. Energy level (n), 2 = 2s or 2p (int)
-    ///   5. l/m label, e.g. "2px" (string)
-    ///
-    /// @arg str_format A string that specifies the output formatting
-    std::vector<std::string> aolabels(std::string str_format = "%2$s%3$d (%4$d%5$s)") const;
-
-    /// Return a vector of AOInfo objects
-    const std::vector<AOInfo>& aoinfo() const;
```

Comment:
Are all these delete functions obsolete or were they reproduced in the python side?

Reply:
They are not used at all, which brings down the codecov.